pkg/operator/ceph/operator.go

```diff
@@ -169,6 +170,11 @@ func (o *Operator) Run() error {
 			close(stopChan)
 			o.clusterController.StopWatch()
 			return nil
+		case err := <-mgrErrorChan:
```

Comment:
Hum shouldn't this be something like `case mgrErrorChan != nil` or something?

Reply:
I do it since err is used later.

```
               case err := <-mgrErrorChan:
                        logger.Errorf("gave up to run the operator. %v", err)
                        close(stopChan)
                        o.clusterController.StopWatch()
                        return err
```

or you mean the code as follows?

```
               case err := <-mgrErrorChan:
                         if err != nil {
                            logger.Errorf("gave up to run the operator. %v", err)
                            close(stopChan)
                            o.clusterController.StopWatch()
                            return err
```

Or any others?

Reply:
Actually just return `err` just like @travisn mentioned below.

Reply:
OK, fixed.

Reply:
Sorry for the back and forth but I'm still not sure this does what we want, this  `err := <-mgrErrorChan:` just populates the channel content in  `err`, right? If so, we still need to verify `err` and then do the cleaning, correct?

@travisn am I missing something?

Reply:
`err := <-mgrErrorChan` will wait for an err to be written to the channel in the other goroutine. If no err is written, this case won't trigger. So it looks correct to me.

Reply:
In the current implementation, `mgrErrorChan` is set iff `o.startManager()` goroutine fails. So when we enter `case err := <-mgrErrorChan:` path, err is always not `nil`.

If inserting `mgrErrorChan <- nil` at the end of `o.startManager()` is better, the code should be like the following.

```
               case err := <-mgrErrorChan:
                         if err != nil {
                            logger.Errorf("gave up to run the operator. %v", err)
                            helper_routine_suggested_by_travis()
                            return err
                         }
               }
```

Both are OK to me.

Reply:
Alright, thanks!